starfish/util/logging.py

```diff
@@ -0,0 +1,50 @@
+import os
+import platform
+from functools import lru_cache
+from json import JSONEncoder
+from subprocess import (
+    call,
+    check_output,
+    PIPE,
+    Popen,
+    STDOUT,
+)
+
+from starfish.types import CORE_DEPENDENCIES
+
+
+@lru_cache(maxsize=1)
+def get_core_dependency_info():
+    dependency_info = dict()
+    for dependency in CORE_DEPENDENCIES:
+        ps = Popen(('pip', 'show', dependency), stdout=PIPE)
+        version = check_output(('grep', 'Version'), stdin=ps.stdout).strip()
+        dependency_info[dependency] = version
+    return dependency_info
+
+
+@lru_cache(maxsize=1)
+def get_git_commit_hash():
+    # First check for git repo
+    if call(["git", "branch"], stderr=STDOUT, stdout=open(os.devnull, 'w')) != 0:
```

Comment:
So this breaks down if someone has a different git repo that pip installs starfish.  Then we would be picking up the hash of the repo that depends on starfish.

I think what we want is to detect that starfish was installed via `pip install -e .` (and its conda equivalent), and only in that case do we log the commit hash.

Reply:
wait you mean only if it wasn't installed with pip or conda then we check for the git hash right?

Reply:
Ah wait I think I understand what you mean. What do you think is the best way to check for that? Checking is the starfish package location doesn't include site-packages?

Reply:
So here's my guess, with basically zero time put into proving its feasibility:

1. get the path of some module in starfish.  resolve it to its `.py` file (https://stackoverflow.com/questions/7162366/get-location-of-the-py-source-file might give you an idea of how to do this, but I would test it out) as opposed to the `.pyc` / `.pyo` file.
2. run `git status --porcelain -- 123 $file`.  this output will be different, depending on whether the file is in a git repo or not.  You may need to `cd` into the directory first though.

Reply:
Also, I would investigate `pkg_resources` to see if you can determine how a package was installed.

Reply:
Having some issues with `git status --porcelain` but alternate simpler idea; what if we just check if a starfish.py is being tracked by git with `git ls-files --error-unmatch starfish.py` and if so do a git descibe

Reply:
> Having some issues with git status --porcelain but alternate simpler idea; what if we just check if a starfish.py is being tracked by git with git ls-files --error-unmatch starfish.py and if so do a git descibe

What issues are you having with `git status --porcelain`?

Reply:
The --porcelain flag doesn't produce any output. Git status does